nemo/utils/exp_manager.py

```diff
@@ -0,0 +1,146 @@
+    _root_dir = root_dir
+    if root_dir is None:
+        _root_dir = str(Path.home() / 'NeMo_experiments')
+    # If the user has already defined a logger for the trainer, use the logger defaults for logging directory
+    if trainer.logger is not None:
+        if trainer.logger.save_dir:
+            if root_dir:
+                raise ValueError(
+                    "The pytorch_lightning trainer that was passed to ExpManager contained a logger, the logger's "
+                    f"save_dir was not None, and root_dir ({root_dir}) was not None. If trainer.logger.save_dir "
+                    "exists, ExpManager will use trainer.logger.save_dir as the logging directory and root_dir "
+                    "must be None."
+                )
+            _root_dir = trainer.logger.save_dir
+        if name:
+            raise ValueError(
+                f"The pytorch_lightning trainer that was passed to ExpManager contained a logger, and name ({name})"
+                " was also passed to ExpManager. If the trainer contains a logger, ExpManager will use "
+                "trainer.logger.name, and name that is passed to ExpManager must be None."
+            )
+        if create_tensorboard_logger:
+            raise ValueError(
+                "The pytorch_lightning trainer that was passed to ExpManager contained a logger, and "
+                "create_tensorboard_logger was set to True. create_tensorboard_logger can only be used if trainer "
+                "does not already have a logger."
+            )
+        name = trainer.logger.name
+        version = trainer.logger.version
+    else:
+        version = None
+        if trainer.is_slurm_managing_tasks:
+            logging.warning("Running on a slurm cluster. Versioning by datetime will not work.")
+        elif trainer.is_global_zero():
+            version = time.strftime('%Y-%m-%d_%H-%M-%S')
+        else:
+            pass  # TODO: grab tm_suffix from sys.argv
+
+        if name is None:
+            name = "default"
+
+    # Always create TensorBoardLogger, so we can retrieve version if running on slurm
+    tensorboard_logger = TensorBoardLogger(save_dir=_root_dir, name=name, version=version)
+    if create_tensorboard_logger:
+        # Attach logger to trainer
+        trainer.configure_logger(tensorboard_logger)
+    if version is None:
+        version = tensorboard_logger.version
+
+    if create_checkpoint_callback:
+        for callback in trainer.callbacks:
+            if isinstance(callback, ModelCheckpoint):
+                raise ValueError(
+                    "The pytorch_lightning trainer that was passed to ExpManager contained a ModelCheckpoint "
+                    "and create_checkpoint_callback was set to True. Please either set create_checkpoint_callback "
+                    "to False, or remove ModelCheckpoint from the lightning trainer"
+                )
+        checkpoint_callback = ModelCheckpoint(save_top_k=3, save_last=True)
```

Comment:
I tried it - it won't work. Strangely ModelCheckpoint callback has to go in trainer(checkpoint_callback=model_checkpoint_callback). If I add it to callbacks list it throws errors. See this https://github.com/PyTorchLightning/pytorch-lightning/issues/1524

Reply:
Thanks for bringing this to my attention. I have hacked around it and it is now working.